sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/ServiceNameCannotBreakDNS.java

```diff
@@ -0,0 +1,47 @@
+package com.mesosphere.sdk.config.validate;
+
+import com.mesosphere.sdk.http.EndpointUtils;
+import com.mesosphere.sdk.offer.LoggingUtils;
+import com.mesosphere.sdk.specification.ServiceSpec;
+import org.slf4j.Logger;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+public class ServiceNameCannotBreakDNS implements ConfigValidator<ServiceSpec> {
+
+    public static final Logger LOGGER = LoggingUtils.getLogger(ServiceNameCannotBreakDNS.class);
+
+
+    private static final int MAX_SERVICE_NAME_LENGTH = 63;
+
+    @Override
+    public Collection<ConfigValidationError> validate(Optional<ServiceSpec> oldConfig, ServiceSpec newConfig) {
+        // Is there a prior config?
+        if (oldConfig.isPresent()) {
+            // There is. We can at most WARN that the name is going to cause truncated DNS.
+            if (!nameWithinSubDomainLimit(newConfig.getName())) {
+                LOGGER.warn("The service name (without slashes) exceeds the maximum size allowed in a DNS subdomain.");
+            }
+
+            return Collections.emptyList();
+        } else {
+            // This is a new deployment, we can fail it if the service name is too large.
+            if (!nameWithinSubDomainLimit(newConfig.getName())) {
+                return Collections.singletonList(ConfigValidationError.valueError(
```

Comment:
If you wanted to be fancy here, the `(without slashes)` in the error could be included only when the service name actually has slashes. But this is such a corner case already...